### ISSUE TITLE
Abnormal CPU activity when filebeat can't connect to Redis instance

### ISSUE THREAD

**Issue by jwadolowski** [elastic/beats#1924]
Filebeat version: 

```
$ filebeat -version
filebeat version 1.2.3 (amd64)
```

OS: 

```
$ cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 6.8 (Santiago)
```

Filebeat config file:

```
$ cat /etc/filebeat/filebeat.yml 
filebeat:
  prospectors: []
  registry_file: "/var/lib/filebeat/registry"
  config_dir: "/etc/filebeat/conf.d"
output:
  redis:
    host: redis.service.consul
    port: 6379
logging:
  to_syslog: false
  to_files: true
  level: info
  files:
    path: "/opt/logs/filebeat"
    name: filebeat.log
    rotateeverybytes: 256000000
    keepfiles: 4
```

When `filebeat` can't connect to defined Redis instance the following warning is printed in logs over and over again:

```
2016-06-28T15:19:43+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:44+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:45+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:46+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:47+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:48+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:49+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:50+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:51+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:52+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:53+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:54+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T15:19:55+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
```

At the same time CPU load significantly increased. Even though retry was set to 1s it shouldn't affect server performance that much. Please find CPU load diagram and `glances` command output below:

![filebeat_abnormal_load_zbx](https://cloud.githubusercontent.com/assets/6334715/16419601/9d8d7190-3d4e-11e6-9832-00164d3c213c.png)

![filebeat_abnormal_load_glances](https://cloud.githubusercontent.com/assets/6334715/16419602/9f74781e-3d4e-11e6-88b6-4ce400653604.png)

**Comment by jwadolowski** [comment]
The load is immediately gone once we restored connectivity to redis.

Log output:

```
2016-06-28T16:01:12+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T16:01:13+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T16:01:14+01:00 WARN Error connecting to Redis (dial tcp: lookup redis.service.consul on 127.0.0.1:53: no such host). Retrying in 1s
2016-06-28T16:01:15+01:00 INFO Events sent: 3
2016-06-28T16:01:15+01:00 INFO Events sent: 2
2016-06-28T16:01:15+01:00 INFO Events sent: 3
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:15+01:00 INFO Events sent: 2048
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:15+01:00 INFO Events sent: 2048
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:15+01:00 INFO Events sent: 2048
2016-06-28T16:01:15+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:21+01:00 INFO Events sent: 1408
2016-06-28T16:01:21+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:29+01:00 INFO Events sent: 3
2016-06-28T16:01:29+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:36+01:00 INFO Events sent: 3
2016-06-28T16:01:36+01:00 INFO Registry file updated. 20 states written.
2016-06-28T16:01:44+01:00 INFO Events sent: 3
2016-06-28T16:01:44+01:00 INFO Registry file updated. 20 states written.
```

**Comment by urso** [comment]
related discuss topic: https://discuss.elastic.co/t/what-would-happend-if-filebeat-couldnt-reach-elasticsearch/53871/4

tl;dr do not use redis output with filebeat 1.x releases. filebeat+redis requires 5.0alpha3 at least.
